src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8333,6 +8376,280 @@ private StatementSyntax ParseLocalDeclarationStatement()
             }
         }
 
+        /// <summary>
+        /// Returns null and resets the pointer if this does not look like a deconstruction-declaration after all.
+        /// </summary>
+        private VariableDeclarationSyntax TryParseDeconstructionDeclaration(bool withEquals)
+        {
+            if (this.CurrentToken.Kind == SyntaxKind.OpenParenToken
+               || (this.CurrentToken.IsVar && this.PeekToken(1).Kind == SyntaxKind.OpenParenToken))
```

Comment:
Yes, it is safe. `PeekToken` returns `EndOfFileToken` after the end of the stream.